src/components/molecules/PollBox/PollBox.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useCallback, useMemo } from "react";
+import { useForm, useFieldArray, Controller } from "react-hook-form";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPaperPlane, faPlus } from "@fortawesome/free-solid-svg-icons";
+import { InputField } from "components/atoms/InputField";
+import { MAX_QUESTIONS_NUMBER } from "settings";
+import "./PollBox.scss";
+
+export type Question = {
+  name: string;
+};
+
+export type PollValues = {
+  topic: string;
+  questions: Question[];
+};
+
+export interface PollBoxProps {
+  onSubmit: (args: PollValues) => void;
+}
+
+const defaultQuestion: Question = { name: "" };
+const defaultValues = {
+  topic: "",
+  questions: [defaultQuestion, defaultQuestion],
+};
+
+export const PollBox: React.FC<PollBoxProps> = ({ onSubmit }) => {
+  const { control, handleSubmit, reset, watch } = useForm<PollValues>({
+    defaultValues,
+  });
+  const { fields, append } = useFieldArray({ name: "questions", control });
+
+  const onCustomSubmit = handleSubmit((data) => {
+    onSubmit(data);
+    reset();
+  });
+
+  const isDisabled = useMemo(() => {
+    const [question1, question2] = watch("questions");
+    return !(watch("topic") && question1.name && question2.name);
+  }, [watch]);
+
+  const addChoice = useCallback(() => append(defaultQuestion), [append]);
+  const showAppend = useCallback(
+    (index) =>
+      index + 1 === fields.length && MAX_QUESTIONS_NUMBER > fields.length,
+    [fields]
+  );
+  const formatPlaceholder = useCallback(
+    (index) =>
+      index === 0
+        ? `Choice ${index + 1} (Max ${MAX_QUESTIONS_NUMBER} choices)`
+        : `Choice ${index + 1}`,
+    []
+  );
+  const renderChoiceFields = useMemo(
+    () =>
+      fields.map((field, index) => (
+        <section className="PollBox__section" key={field.id}>
+          <Controller
+            as={
+              <InputField
+                containerClassName="PollBox__input"
+                autoComplete="off"
+                placeholder={formatPlaceholder(index)}
+              />
+            }
+            control={control}
```

Comment:
What does the `Controller`/`control` part of this do?

Reply:
control object is from invoking useForm
https://react-hook-form.com/api/usecontroller/controller

Reply:
The `Controller` docs you link talk about using `render`, but doesn't seem to mention anything about `as`/etc?

Is the code here correct? Or are there other docs that described the way you are using it here? Etc?

Reply:
@mike-lvov @sunny-viktoryia  As mentioned in @mike-lvov's PR, i'm not sure that `InputField`'s `ref` is connected to the right thing. I expect it may need to be added to the `input` rather than the `div`. Which based on how the docs for `Controller` describe it working, may impact this here too.

Reply:
my bad. Seems I might have misunderstood the intention of adding `ref` to `InputField` when I was adding it.
created a pull request that moves ref from `div` to `input`.
https://github.com/sparkletown/sparkle/pull/1353

sorry for the inconveniences caused.

Reply:
@sunny-viktoryia No worries. Thanks for making the fix + pointing out where it was originally discussed (I thought my memory was going crazy :p)